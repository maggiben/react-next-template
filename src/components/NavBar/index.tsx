import Image from "next/image";
import { Heading, Grid, GridItem } from '@fravega-it/bumeran-ds-fvg';
import { useTranslation } from 'react-i18next';
import React from 'react';
import logo from '@images/logo-fravega.png';
import styled from 'styled-components';

const Header = styled.div`
  height: 80px;
  border-width: 1px;
`;

const Logo = styled.div`
  display: flex;
  height: 80px;
  justify-content: start;
  align-items: center;
`;

const User = styled.div`
  display: flex;
  height: 80px;
  justify-content: start;
  align-items: center;
`;

export default function NavBar(): JSX.Element {
  const { t } = useTranslation();
  return (
    <Header>
      <Grid>
        <GridItem xs={8} justifySelf="stretch" alignSelf="center">
          <Logo>
            <Image priority src={logo} alt="BÃºmeran logo" style={{marginRight: '12px'}} width={130}/>
            <Heading color="neutral" size="s">{t('client finder')}</Heading>
          </Logo>
        </GridItem>
        <GridItem xs={4} justifySelf="end" alignSelf="center">
          <User />
        </GridItem>
      </Grid>
    </Header>
  );
}
