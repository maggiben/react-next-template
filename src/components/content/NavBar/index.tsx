import { useEffect, useState } from 'react';
import { SearchIcon, UserIcon, Heading, IconButton } from '@fravega-it/bumeran-ds-fvg'
import React from 'react';
import styled from "styled-components";

const Header = styled.div`
  display: flex;
  height: 80px;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  border-width: 1px;
  background-color: ${({ theme }) => theme.colors.violet[400]};
`;

const Logo = styled.div`
  display: flex;
  height: 80px;
  justify-content: start;
  align-items: center;
`;

const User = styled.div`
  display: flex;
  height: 80px;
  justify-content: start;
  align-items: center;
`;

export default function NavBar(): JSX.Element {
  return (
    <Header>
      <Logo>
      <SearchIcon size="l" color="white" colorTone="600" />
      <Heading color="white">Buscador de Clientes</Heading>
      </Logo>
      <User>
        {/* <IconButton icon={<UserIcon size="l" color="white" colorTone="600" />} /> */}
      </User>
    </Header>
  );
}
