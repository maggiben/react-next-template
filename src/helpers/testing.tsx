import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};