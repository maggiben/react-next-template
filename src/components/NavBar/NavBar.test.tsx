import React from 'react';
import { renderer } from '@helpers/testing';
import '@testing-library/jest-dom';
import NavBar from './';
import { expect } from '@jest/globals';

describe('NavBar', () => {
  it('renders the NavBar component', () => {
    const tree = renderer.create(<NavBar />);
    expect(tree).toMatchSnapshot();
  });
});
