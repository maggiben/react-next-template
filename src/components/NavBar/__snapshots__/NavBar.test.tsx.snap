// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders the NavBar component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="NavBar__Header-sc-8wifgb-0 gdDcDj"
      >
        <div
          class="sc-jsJBEP gDratY"
        >
          <div
            class="sc-eeDRCY uWSV"
          >
            <div
              class="NavBar__Logo-sc-8wifgb-1 jlULcu"
            >
              <a
                href="/"
              >
                <img
                  alt="Buscador de Clientes"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="24"
                  src="/_next/image?url=%2Fstatic%2Fimages%2Flogo-fravega.png&w=384&q=75"
                  srcset="/_next/image?url=%2Fstatic%2Fimages%2Flogo-fravega.png&w=256&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Flogo-fravega.png&w=384&q=75 2x"
                  style="color: transparent; margin-right: 12px;"
                  width="130"
                />
              </a>
              <div
                aria-level="5"
                class="sc-kAyceB iIRIPg"
                color="neutral"
                role="heading"
              >
                Buscador de Clientes
              </div>
            </div>
          </div>
          <div
            class="sc-eeDRCY fHgkUB"
          >
            <div
              class="NavBar__User-sc-8wifgb-2 hSvnZj"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="NavBar__Header-sc-8wifgb-0 gdDcDj"
    >
      <div
        class="sc-jsJBEP gDratY"
      >
        <div
          class="sc-eeDRCY uWSV"
        >
          <div
            class="NavBar__Logo-sc-8wifgb-1 jlULcu"
          >
            <a
              href="/"
            >
              <img
                alt="Buscador de Clientes"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="24"
                src="/_next/image?url=%2Fstatic%2Fimages%2Flogo-fravega.png&w=384&q=75"
                srcset="/_next/image?url=%2Fstatic%2Fimages%2Flogo-fravega.png&w=256&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Flogo-fravega.png&w=384&q=75 2x"
                style="color: transparent; margin-right: 12px;"
                width="130"
              />
            </a>
            <div
              aria-level="5"
              class="sc-kAyceB iIRIPg"
              color="neutral"
              role="heading"
            >
              Buscador de Clientes
            </div>
          </div>
        </div>
        <div
          class="sc-eeDRCY fHgkUB"
        >
          <div
            class="NavBar__User-sc-8wifgb-2 hSvnZj"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
